src/SFA.DAS.EmployerApprenticeshipsService.Application/Queries/GetAccountTransferRole/GetAccountTransferRoleHandler.cs

```diff
@@ -0,0 +1,68 @@
+using System.Data;
+using System.Data.Entity;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using MediatR;
+using SFA.DAS.EAS.Application.Data;
+using SFA.DAS.EAS.Application.Dtos;
+using SFA.DAS.EAS.Application.Validation;
+using SFA.DAS.EAS.Domain.Models.TransferConnections;
+using SFA.DAS.HashingService;
+
+namespace SFA.DAS.EAS.Application.Queries.GetAccountTransferRole
+{
+    public class GetAccountTransferRoleHandler : IAsyncRequestHandler<GetAccountTransferRoleQuery, GetAccountTransferRoleResponse>
+    {
+        private readonly EmployerAccountDbContext _db;
+        private readonly IHashingService _hashingService;
+
+        public GetAccountTransferRoleHandler(EmployerAccountDbContext db, IHashingService hashingService)
+        {
+            _db = db;
+            _hashingService = hashingService;
+        }
+
+        public async Task<GetAccountTransferRoleResponse> Handle(GetAccountTransferRoleQuery message)
+        {
+            var response = new GetAccountTransferRoleResponse
+            {
+                AccountId = _hashingService.DecodeValue(message.AccountHashedId)
+            };
+
+            var parameters = new SqlParameter[]
+            {
+                new SqlParameter("accountId", SqlDbType.BigInt) {Value = response.AccountId},
+                new SqlParameter("sentPending", SqlDbType.Int){Direction = ParameterDirection.Output},
+                new SqlParameter("sentApproved", SqlDbType.Int){Direction = ParameterDirection.Output},
+                new SqlParameter("receivedPending", SqlDbType.Int){Direction = ParameterDirection.Output},
+                new SqlParameter("receivedApproved", SqlDbType.Int){Direction = ParameterDirection.Output},
+            };
+
+            await _db.Database.ExecuteSqlCommandAsync($@"
+                SELECT	@sentPending = ISNULL(SUM(CASE WHEN tci.SenderAccountId = @accountId and tci.Status = 1 THEN 1 ELSE 0 END), 0),
+		                @sentApproved = ISNULL(SUM(CASE WHEN tci.SenderAccountId = @accountId and tci.Status = 2 THEN 1 ELSE 0 END), 0),
+		                @receivedPending = ISNULL(SUM(CASE WHEN tci.ReceiverAccountId = @accountId and tci.Status = 1 THEN 1 ELSE 0 END), 0),
+		                @receivedApproved = ISNULL(SUM(CASE WHEN tci.ReceiverAccountId = @accountId and tci.Status = 2 THEN 1 ELSE 0 END), 0)
+                FROM    employer_account.TransferConnectionInvitation as tci
+                WHERE	(tci.SenderAccountId = @accountId or tci.ReceiverAccountId = @accountId) and tci.Status != {(int)TransferConnectionInvitationStatus.Rejected}",
+                parameters);
+
+            response.Sender = new TransferRoleTypeInfo
```

Comment:
Although the roles/counts *might* be needed in future they aren't needed to satisfy this story's requirements. Could you simplify your response to just having two bool properties e.g. `IsPendingReceiver` & `IsApprovedReceiver`?